packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWTypeaheadSelectList/CWTypeaheadSelectList.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import Select, { components } from 'react-select';
+import { ComponentType } from '../../types';
+import './CWTypeaheadSelectList.scss';
+import { CWIcon } from '../../cw_icons/cw_icon';
+
+type SelectListOption = {
+  value: any;
+  label: string;
+};
+
+type TypeaheadSelectListProps = {
+  options: SelectListOption[];
+  defaultValue: SelectListOption;
+  placeholder: string;
+  isDisabled?: boolean;
+};
+
+const DropdownIndicator = (props) => {
+  const isMenuOpen = props.selectProps.menuIsOpen;
+
+  return (
+    <components.DropdownIndicator {...props}>
+      {isMenuOpen ? (
+        <CWIcon iconName="chevronUp" iconSize="small" />
+      ) : (
+        <CWIcon iconName="chevronDown" iconSize="small" />
+      )}
+    </components.DropdownIndicator>
+  );
+};
+
+const Option = (props) => {
+  return (
+    <components.Option {...props}>
+      <div className="text-container">{props.label}</div>
+    </components.Option>
+  );
+};
+
+export const CWTypeaheadSelectList = ({
+  options,
+  defaultValue,
+  placeholder,
+  isDisabled = false,
+}: TypeaheadSelectListProps) => {
+  return (
+    <div className={ComponentType.TypeaheadSelectList}>
+      <Select
```

Comment:
Can we add a fullwidth version, like a `fullWidth` prop? I think there will be some places we would use that

Reply:
That may be useful, but in the spirit of not adding more than we might need, I'll wait for design to specify the assorted widths.